packages/sources/por-address-list/src/transport/utils.ts

```diff
@@ -5,22 +5,34 @@ export const fetchAddressList = async (
   latestBlockNum: number,
   confirmations = 0,
   batchSize = 10,
+  batchGroupSize = 10,
 ): Promise<string[]> => {
   const blockTag = latestBlockNum - confirmations
   const numAddresses = await addressManager.getPoRAddressListLength({
     blockTag,
   })
-  let addresses: string[] = []
+  let totalRequestedAddressesCount = 0
   let startIdx = ethers.BigNumber.from(0)
-  while (ethers.BigNumber.from(addresses.length).lt(numAddresses)) {
+  const addresses: string[] = []
+  let batchRequests: Promise<string>[] = []
```

Comment:
I believe `getPoRAddressList` would return a `string[]` so I think this should be 
```suggestion
  let batchRequests: Promise<string[]>[] = []
```